spec/controllers/assignments_controller_spec.rb

```diff
@@ -415,100 +436,276 @@
     end
   end
 
+  context '#new' do
+    context 'as an admin' do
+      shared_examples 'assignment_new_success' do
+        it 'responds with a success' do
+          expect(response).to have_http_status :success
```

Comment:
I've been trying to use [`assert_response`](https://apidock.com/rails/v6.0.0/ActionDispatch/Assertions/ResponseAssertions/assert_response) more in testing status codes.

Reply:
I can do that but I prefer to use rspec matchers for rspec tests as much as possible because in general it gives better error messages. Also I'm pretty sure `have_http_status` calls `assert_response` so the mechanism should be the same.

Reply:
Okay that sounds good, thanks for letting me know.